src/Uno.UI.Runtime.Skia.X11/X11XamlRootHost.cs

```diff
@@ -534,10 +537,57 @@ private bool IsOpenGLSupported(IntPtr display)
 		}
 	}
 
-	void IXamlRootHost.InvalidateRender() => _renderDirty = true;
-
 	UIElement? IXamlRootHost.RootElement => _window.RootElement;
 
+	// running XConfigureEvent and rendering callbacks on a timer is necessary to avoid freezing in certain situations
+	// where we get spammed with these events. Most notably, when resizing a window by dragging an edge, we'll get spammed
+	// with XConfigureEvents.
+	void IXamlRootHost.InvalidateRender()
+	{
+		if (Interlocked.Exchange(ref _renderScheduled, 1) == 0)
+		{
+			// Don't use ticks, which seem to mess things up for some reason
+			var now = _stopwatch.ElapsedMilliseconds;
+			var delta = now - Interlocked.Exchange(ref _lastRenderTime, now);
+			if (delta > TimeSpan.FromSeconds(1.0 / X11ApplicationHost.RenderFrameRate).TotalMilliseconds)
+			{
+				QueueAction(this, () =>
+				{
+					_renderScheduled = 0;
+					_renderer?.Render();
+				});
+			}
+			else
+			{
+				_renderTimer.Interval = TimeSpan.FromTicks(delta);
+				_renderTimer.Start();
```

Comment:
If I understand correctly, we're setting up the interval to be the time since the last update? Aren't we trying to render immediately, or at least at the FPS interval?

Reply:
We're rendering immediately if the last frame was more than 1/60 secs ago. If the last frame is within the last 1/60 secs, we wait the remainder of this 1/60 secs "time slice" before rendering again, basically capping the frame rate at 60.